### PR TITLE
finalize-app

### PR DESCRIPTION
**Pull request by migo1**
## Design

- Follow design guidelines including colors, typography, layout, and space between elements.
- Credit the original design idea by Gregoire Vella on Behance.

### Home Page

- A simple page with the app's name and links to the sign-up and log-in pages.

### Sign-up and Log-in Pages

- User registration with full name, email, and password (all mandatory).
- User login with email and password.
- Unauthorized users can't access pages requiring login.

### Categories Page

- Users see the categories page after login.
- Each category displays its name, icon, and the total amount of associated transactions.
- Clicking a category item navigates to the transactions page for that category.
- "Add a new category" button for creating new categories.

### Transactions Page

- Displays transactions for a selected category, ordered by the most recent.
- Shows the total amount for the category at the top.
- "Add a new transaction" button for creating new transactions.
- The "Back" button navigates to the home page.

### "Add a New Category" Page

- User fills out a form to create a new category, specifying name and icon (both mandatory).
- The "Save" button creates the category and redirects to the home page.
- The "Back" button navigates to the home page.

### "Add a New Transaction" Page

- User fills out a form to create a new transaction with name, amount, and categories (mandatory).
- The "Save" button creates the transaction and navigates to the transactions page for that category.
- The "Back" button navigates to the transactions page for that category.

### CODE REVIEW

File: app/views/categories/show.html.erb [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <ul class="px-2">
   <% @category.purchases.each do |purchase| %>
    <a href="<%= category_purchase_path(@category, purchase) %>" >
-    <li class="py-3 bg-white flex justify-between px-2">
+    <li class="py-3 bg-white flex justify-between px-2 mb-2">
       <div>
         <p class="font-bold">
           <%= purchase.name %>
```
Comment (mikethreels) [inline app/views/categories/show.html.erb]: - [ ] When adding items to the list the total doesn't add up to the total of the items.
image

File: app/models/category.rb [hunk 2]
```diff
@@ -6,4 +6,16 @@ class Category < ApplicationRecord
...
   validates :image, presence: true
+
+  def total_sum
+    purchases.sum(:amount)
+  end
+
+  before_destroy :delete_associated_purchases
+
+  private
+
```
Comment (mikethreels) [inline app/models/category.rb:17]: - [ ] in the "add a new transaction" page the user should get the option to choose the category as well according to requirements please make sure to add this option.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Migot, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job creating the application with all the associations.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/categories/show.html.erb (hunk 1 above)
Comment (migo1) [reply app/views/categories/show.html.erb]: @mikethreels noted, the current branch isn't deployed I'll deploy it

File: app/models/category.rb (hunk 2 above)
Comment (migo1) [reply app/models/category.rb:17]: @mikethreels noted

**Review by BenMukebo — APPROVED** [review]
# Status: Approved  :champagne:  ✔️:confetti_ball: 

## Hi @migo1, 

### **Congratulations! 🎉 Project Approved ✅**

Good job so far. Well Done.👍🏻 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight 🎉
- Git flow is followed. :heavy_check_mark:
- You have correctly setup devise for authentication. ✔️ 
- The database schema is reflecting the ERD diagram. 👍🏻 
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Good presentation video. ✔️ 
- Your project was deployed properly. heavy_check_mark
- Your design looks awesome. keep rocking :rocket: 

## Optional suggestions

- [OPIONAL ] It would be much better to add more input-modal validation. This will increase the app security and will make sure that anyone with bad intentions cannot compromise their app. ✍🏻 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/categories/show.html.erb [hunk 3]
```diff
@@ -8,9 +8,9 @@
...
 
     <ul class="px-2">
-  <% @category.purchases.each do |purchase| %>
+  <% @purchases.each do |purchase| %>
    <a href="<%= category_purchase_path(@category, purchase) %>" >
-    <li class="py-3 bg-white flex justify-between px-2">
+    <li class="py-3 bg-white flex justify-between px-2 mb-2">
       <div>
         <p class="font-bold">
           <%= purchase.name %>
```
Comment (BenMukebo) [inline app/views/categories/show.html.erb:16]: - [OPTIONAL] Great job so far 👏🏻 , however For a given category, the list of transactions should be presented in order by the most recent. This means, the cat2 I created after cat1 👇🏻 should be at the top and cat1 the last one. Please address this 🧑‍🔧 

image


- [OPTIONAL] I would prefer for the `New Purchase` button design to look similar to the `New Category` button, eg: border, heght, ...

image

Comment (migo1) [reply app/views/categories/show.html.erb:16]: @BenMukebo noted, thanks for the review I'll work on the requested changes

**Pull request merged**